src/adapters/direction_api.js

```diff
@@ -74,10 +74,16 @@ export default class DirectionApi {
     try {
       response = await Ajax.get(directionsUrl, directionsParams, { timeout });
     } catch (e) {
-      return;
+      if (Number.isInteger(e) && e >= 400 && e < 600) {
+        // Use the error codes 4xx and 5xx to display different error messages
+        response = { data: null, error: e };
```

Comment:
I would just return here and below, instead of rewriting the response and risking it to overwritten below.